src/providers/auth.service.ts

```diff
@@ -121,4 +122,27 @@ export class AuthService {
       });
     });
   }
+
+  /**
+   * Delete user
+   *
+   * @returns {Observable}
+   */
+  public deleteUser() {
+    return new Observable((observer) => {
+      this.authHttp.delete(environment.server_url + "/api/user")
+        .subscribe(
+          data => {
+            observer.next(data);
+            observer.complete();
+          },
+          err => {
+            observer.next(err);
+            observer.complete();
+          },
+          () => {
+            observer.complete();
```

Comment:
Fair enough, however we have not been using on complete so just use on next and on error.